core/src/wallet/tezos/explorers/ExternalTezosLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,257 @@
+                const std::shared_ptr<HttpClient> &http,
+                const api::TezosLikeNetworkParameters &parameters,
+                const std::shared_ptr<api::DynamicObject> &configuration) :
+                DedicatedContext(context),
+                TezosLikeBlockchainExplorer(configuration, {api::Configuration::BLOCKCHAIN_EXPLORER_API_ENDPOINT}) {
+            _http = http;
+            _parameters = parameters;
+        }
+
+
+        Future<std::shared_ptr<BigInt>>
+        ExternalTezosLikeBlockchainExplorer::getBalance(const std::vector<TezosLikeKeychain::Address> &addresses) {
+            auto size = addresses.size();
+            if (size != 1) {
+                throw make_exception(api::ErrorCode::INVALID_ARGUMENT,
+                                     "Can only get balance of 1 address from Tezos Node, but got {} addresses",
+                                     addresses.size());
+            }
+            std::string addressesStr = addresses[0]->toString();
+            return getHelper(fmt::format("account/{}", addressesStr),
+                             "total_balance",
+                             std::unordered_map<std::string, std::string>{}
+            );
+        }
+
+        Future<std::shared_ptr<BigInt>>
+        ExternalTezosLikeBlockchainExplorer::getFees() {
+            bool parseNumbersAsString = true;
+            return _http->GET("block/head")
+                    .json(parseNumbersAsString).mapPtr<BigInt>(getContext(), [](const HttpRequest::JsonResult &result) {
+                        auto &json = *std::get<1>(result);
+                        //Is there a fees field ?
+                        if (!json.IsObject() || !json.HasMember("fees") ||
+                            !json["fees"].IsString()) {
+                            throw make_exception(api::ErrorCode::HTTP_ERROR,
+                                                 "Failed to get fees from network, no (or malformed) field \"result\" in response");
+                        }
+                        std::string fees = json["fees"].GetString();
+                        // Sometimes network is sending 0 for fees
+                        if (fees == "0") {
+                            fees = api::TezosConfigurationDefaults::TEZOS_DEFAULT_FEES;
+                        }
+                        return std::make_shared<BigInt>(fees);
+                    });
+        }
+
+        Future<String>
+        ExternalTezosLikeBlockchainExplorer::pushLedgerApiTransaction(const std::vector<uint8_t> &transaction) {
+            std::stringstream body;
+            body << '"' << hex::toString(transaction) << '"';
+            auto bodyString = body.str();
+            return _http->POST("/injection/operation?chain=main",
+                               std::vector<uint8_t>(bodyString.begin(), bodyString.end()),
+                               std::unordered_map<std::string, std::string>{},
+                               "https://mainnet.tezrpc.me")
```

Comment:
Why the url isn't store the same way than the others ?

Reply:
It could, it was just to avoid creating dozens of urls specially that it is the only place we use it !